h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -73,11 +78,56 @@ class SearchBarController extends Controller {
     };
 
     var openDropdown = () => {
+      clearActiveDropdownItem();
+
+      // Reset dropdown items to not hidden
+      Object.keys(this._dropdownItems).forEach(function(key) {
+        this._dropdownItems[key].classList.remove('is-hidden');
+      }.bind(this));
+
       this.setState({open: true});
+
       this._input.addEventListener('keydown', setupListenerKeys,
         true /*capture*/);
     };
 
+    var getVisibleDropdownItems = () => {
+      return this._dropdown.querySelectorAll('li:not(.is-hidden)');
+    };
+
+    var maybeOpenOrCloseDropdown = () => {
+      var word = this._input.value.trim();
+
+      if (word.length > 0) {
+        openDropdown();
+
+        // if the word has a ':' don't show dropdown
+        if (word.indexOf(':') > -1) {
+          closeDropdown();
+        }
```

Comment:
It might be good to simplify this function a little so that it always either calls `openDropdown()` once or calls `closeDropdown()` once. At the moment I think it can under certain circumstances call `openDropdown()` and then `closeDropdown()`.